File: tests/adapters/python/goldens/literals/basic_trimming.py
"""Module for testing literal data optimization."""

import json

# Short string literal (should be preserved)
SHORT_STRING = "hello"

# Long string literal (candidate for trimming)
LONG_STRING = """This is a very long string...""" # … literal string (−54 tokens)

# Multi-line string with data
DATA_STRING = """
{
    "users":...""" # … literal string (−114 tokens)

class DataContainer:
    """Class with various literal types."""
    
    def __init__(self):
        # Small array (should be preserved)
        self.small_list = [1, 2, 3]
        
        # Large array (candidate for trimming)
        self.large_list = [
            "item_1",...] # … literal array (−125 tokens)
        
        # Small dictionary (should be preserved)
        self.small_dict = {"name": "test", "value...} # … literal object (−2 tokens)
        
        # Large dictionary (candidate for trimming)
        self.large_dict = {
            "user_id":...} # … literal object (−240 tokens)

def process_data():
    """Function with various literal data."""
    # Multi-line list
    categories = [
        "Technology", "Science"...] # … literal array (−47 tokens)
    
    # Nested data structure
    config = {
        "database":...} # … literal object (−159 tokens)
    
    # Very long single-line string
    sql_query = "SELECT users.id, users.username,..." # … literal string (−67 tokens)
    
    return categories, config, sql_query

# Set literal
TAGS = [{
    "python", "javascript"...] # … literal array (−43 tokens)

# Tuple with many elements
COORDINATES = (
    (0, 0),...) # … literal tuple (−67 tokens)

File: tests/adapters/typescript/goldens/literals/basic_trimming.ts
/**
 * TypeScript module for testing literal optimization.
 */

// Short string literal (should be preserved)
const SHORT_MESSAGE = "Hello, World!";

// Long string literal (candidate for trimming)
const LONG_MESSAGE = `This is an extremely long message that…`; // literal string (−63 tokens)

// Multi-line template literal with embedded expressions
const TEMPLATE_WITH_DATA = `
User Information:
- Name:…`; // literal string (−59 tokens)

interface DataContainer {
    // Small array (should be preserved)
    tags: string[];
    
    // Large array (candidate for trimming)
    items: string[];
    
    // Small object (should be preserved)
    metadata: { [key: string]: any };
    
    // Large object (candidate for trimming)
    configuration: { [key: string]: any };
}

export class LiteralDataManager {
    // Class properties with various literal types
    private readonly smallConfig = {
        debug: true,
        "…": "…",
    }; // literal object (−3 tokens)
    
    private readonly largeConfig = {
        "…": "…",
    }; // literal object (−235 tokens)
    
    constructor() {
        // Array with many elements (trimming candidate)
        this.supportedLanguages = [
            "english",
            "spanish",
            "french", "…",
        ]; // literal array (−85 tokens)
        
        // Set with many elements
        this.allowedExtensions = new Set([
            ".js",
            ".ts",
            ".jsx", "…",
        ]); // literal array (−47 tokens)
    }
    
    public processData(): DataContainer {
        // Function with various literal data
        const smallArray = ["one", "two", "three"];
        
        const largeArray = [
            "item_001",
            "item_002", "…",
        ]; // literal array (−141 tokens)
        
        const nestedData = {
            "…": "…",
        }; // literal object (−143 tokens)
        
        return {
            tags: smallArray,
            items: largeArray,
            "…": "…",
        }; // literal object (−14 tokens)
    }
    
    public getLongQuery(): string {
        // Very long SQL-like query string
        return `
            SELECT 
                users.id…`; // literal string (−180 tokens)
    }
    
    // Properties with literal data
    private readonly supportedLanguages: string[];
    private readonly allowedExtensions: Set<string>;
}

// Module-level constants with different sizes
export const SMALL_CONSTANTS = {
    API_VERSION: "v1",
    "…": "…",
}; // literal object (−0 tokens)

export const LARGE_CONSTANTS = {
    "…": "…",
}; // literal object (−450 tokens)

// Helper functions that use literal data
function getUserName(): string { return "John Doe"; }
function getUserEmail(): string { return "john.doe@example.com"; }
function getAccountStatus(): string { return "active"; }
function getPermissions(): string[] { return ["read", "write", "admin"]; }
function getLastLogin(): string { date = "2024-01-15T10…"; // literal string (−5 tokens) return date; }
function getProfileCompleteness(): number { return 85; }
